### PR TITLE
Avoid potential overflow during multiplication

### PR DESCRIPTION
**Pull request by ssoloff**
## Overview

Fixes potential overflows in multiplicative expressions, as identified by LGTM.

Based on the actual range of the operands involved in these expressions, I don't think any of them would actually overflow.  Nonetheless, these changes ensure the wider type is used consistently throughout the expression.

I believe in all cases there should be no loss in precision.  However, please point out any expression where you think this assertion may not hold.

## Functional Changes

None.

## Manual Testing Performed

None.

### CODE REVIEW

**Comment by ron-murhammer** [comment]
This pull request **fixes 30 alerts** when merging ed4d0c006422087ca8d5dfa3c2bfa38a575adcce into 82e0dc189c0cc787e634ad741831528d65516cb4 - [view on LGTM.com](https://lgtm.com/projects/g/triplea-game/triplea/rev/pr-a7a861d1b4e2a4936ee5d3bbc3e6343e199d5957)

**fixed alerts:**

* 30 for Result of multiplication cast to wider type

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by ron-murhammer** [comment]
This pull request **fixes 31 alerts** when merging ab800c4d4d45dd4786173a87353c30048c72d22a into 82e0dc189c0cc787e634ad741831528d65516cb4 - [view on LGTM.com](https://lgtm.com/projects/g/triplea-game/triplea/rev/pr-f9a8e11eb6e26e0697d28221bb0a1f57d839a321)

**fixed alerts:**

* 31 for Result of multiplication cast to wider type

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by ssoloff** [comment]
@RoiEXLab 

> I'm interested to hear about the details for the cases where the simple end result is being cast.
I don't really get why this is necessary.

I'm assuming you're talking about something like this:

```java
return (double) production * Properties.getPuMultiplier(data);
```

The precedence of the cast operator is higher than that of the multiplication operator, so this is equivalent to

```java
return ((double) production) * Properties.getPuMultiplier(data);
```

Not

```java
return (double) (production * Properties.getPuMultiplier(data));
```

File: game-core/src/main/java/games/strategy/engine/random/RandomStatsDetails.java [hunk 1]
```diff
@@ -68,7 +68,7 @@ private static DiceStatistic getDiceStatistic(final IntegerMap<Integer> stats, f
       double sumOfSquaredMeanDeviations = 0;
       // TODO: does this need to be updated to take data.getDiceSides() ?
       for (int i = 1; i <= diceSides; i++) {
-        sumOfSquaredMeanDeviations += (stats.getInt(i) - (localTotal / diceSides))
+        sumOfSquaredMeanDeviations += (double) (stats.getInt(i) - (localTotal / diceSides))
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/engine/random/RandomStatsDetails.java]: @ssoloff Yes and no.
The code snippet you presented was one of them, but this is another occurence.

Comment (ssoloff) [reply game-core/src/main/java/games/strategy/engine/random/RandomStatsDetails.java]: That's the same thing, right?

```java
sumOfSquaredMeanDeviations += (double) (stats.getInt(i) - (localTotal / diceSides)) * (stats.getInt(i) - (localTotal / diceSides));
sumOfSquaredMeanDeviations += (double) (i) * (i);
sumOfSquaredMeanDeviations += ((double) i) * i;
```

That one might have been clearer by first extracting the mean deviation term and then summing the square:

```java
final double meanDeviation = stats.getInt(i) - (localTotal / diceSides);
sumOfSquaredMeanDeviations += meanDeviation * meanDeviation;
```

Comment (RoiEXLab) [reply game-core/src/main/java/games/strategy/engine/random/RandomStatsDetails.java]: @ssoloff Yes, my bad. I didn't realize that the multiplication was split over 2 lines, thus my confusion.
Thanks for clarifying. Makes sense now 👍

**Pull request merged**